### PR TITLE
More complete FAQ and context about releasing next patchlevel release

### PR DESCRIPTION
**Pull request by potiuk**
Following the questions raised in:
* https://lists.apache.org/thread/6nkwonmn4gp5o6pw2m9yzo4y8qhst0md

And spun-off explanation in:
* https://lists.apache.org/list.html?dev@airflow.apache.org

This PR adds more commplete FAQ and context describing the process we follow to release patchlevel release and describes roles involved in the process and their responsibilities, as well as explains our approach to SemVer/Breaking changes when we are choosing which changes should be included and whether they are breaking or not.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by ephraimbuddy — APPROVED** [review]

File: dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+We have a well defined and well tested process that we follow.,
+We follow it since the beginning of Airflow 2.0 actually - it's been also done the same in 1.10 -
+but with some variations, we do it the same way since the beginning of 2.0,
+but it has been refined and improved over time - by those who volunteered their time in the release
+process (a lot of ad-hoc discussion have been traditionally happening in #release-management slack channel)
+and we continue to do so.
+
+The succinct form of it is described in a prominent place in our most important
+[README](../README.md#what-goes-into-the-next-release).
```
Comment (ephraimbuddy) [inline dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md:48]: I think this needs o be changed?

File: dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+the likelihood of breaking our user's workflow is low.
+
+# Do we intentionally skip over some changes when releasing a patchlevel release?
+
+Skipping is not intentional because we never "skip" things when cherry-picking, It's **reverse** -
+those maintainer who think that certain bug fixes (or internal changes or sometimes even feature changes
+that we classify really as "bugfix" SHOULD intentionally mark those PRs they want with the next `patch-level`
+milestone to be included.  So there is no skipping, if maintainer did not deliberately mark PR as
+upcoming milestone, it will just not be included (not skipped). By default all the changes merged to main
+are included in the next `minor` release. See [README](../README.md#what-goes-into-the-next-release) for
```
Comment (ephraimbuddy) [inline dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md:129]: This too

File: dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md (hunk 1 above)
Comment (potiuk) [reply dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md:48]: Why ? I think it's good link, we still want to describe a much shorter version of it in the README ? or I miss what needs to be changed :) ?

Comment (ephraimbuddy) [reply dev/WHAT_GOES_INTO_THE_NEXT_RELEASE.md:48]: I was thinking we are removing the one on README and having this one replace it. Obviously that's not the case, resolving this

**Pull request merged**